source/extensions/quic_listeners/quiche/active_quic_listener.cc

```diff
@@ -94,16 +100,24 @@ void ActiveQuicListener::onData(Network::UdpRecvData& data) {
 }
 
 void ActiveQuicListener::onReadReady() {
+  if (!enabled()) {
+    ENVOY_LOG(trace, "Quic listener {}: runtime disabled", config_.name());
+    return;
+  }
```

Comment:
I think we can probably just guard the read ready case? I will defer to @danzh2010 on this.

Reply:
Yes, guarding onData()not necessary. 

On the other hand, do we want the kill switch to gracefully close all new and existing connections before blocking all the incoming packets? If no, I think guarding at onReadReady() and onWriteReady() is sufficient. Otherwise we probably need to use QUICHE interface in QuicDispatcher to reset all the quic connections. @mattklein123 @alyssawilk